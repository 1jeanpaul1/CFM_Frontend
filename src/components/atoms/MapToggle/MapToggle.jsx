import PropTypes from 'prop-types';
import { Button } from '@mui/material';
import {
  MapOutlined as MapIcon,
  FormatListBulletedOutlined as ListIcon,
} from '@mui/icons-material';

export default function MapToggle({ currentView, setView }) {
  return (
    <Button
      fullWidth
      startIcon={
        currentView === MapToggle.view.map ? <ListIcon /> : <MapIcon />
      }
      sx={{
        position: 'fixed',
        bottom: 0,
        zIndex: 999,
        height: 60,
        backgroundColor: '#fff',
        border: 'none',
        borderRadius: 3,
        borderBottomLeftRadius: 0,
        borderBottomRightRadius: 0,
        boxShadow: '-2px 0px 4px rgb(0 0 0 / 20%)',
        justifyContent: 'left',
        padding: 5,
        fontWeight: 500,
        textTransform: 'none',
        ':hover': { backgroundColor: '#fff' },
      }}
      onClick={() =>
        setView(
          currentView === MapToggle.view.map
            ? MapToggle.view.list
            : MapToggle.view.map
        )
      }
    >
      {currentView === MapToggle.view.map ? 'List View' : 'Map View'}
    </Button>
  );
}
MapToggle.propTypes = {
  currentView: PropTypes.symbol,
  setView: PropTypes.func,
};

MapToggle.view = Object.freeze({
  map: Symbol(0),
  list: Symbol(1),
});
