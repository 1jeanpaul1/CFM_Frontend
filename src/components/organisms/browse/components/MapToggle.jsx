import PropTypes from 'prop-types';
import { IconButton, Typography } from '@mui/material';
import {
  MapOutlined as MapIcon,
  FormatListBulletedOutlined as ListIcon,
} from '@mui/icons-material';

export default function MapToggle({ currentView, setView }) {
  return (
    <IconButton
      onClick={() =>
        setView(
          currentView === MapToggle.view.map
            ? MapToggle.view.list
            : MapToggle.view.map
        )
      }
      sx={{ color: 'white' }}
    >
      {currentView === MapToggle.view.map ? <ListIcon /> : <MapIcon />}
      <Typography sx={{ minWidth: '2.5rem' }}>
        {currentView === MapToggle.view.map ? 'List' : 'Map'}
      </Typography>
    </IconButton>
  );
}
MapToggle.propTypes = PropTypes.exact({
  currentView: PropTypes.symbol,
  setView: PropTypes.func,
}).isRequired;

MapToggle.view = Object.freeze({
  map: Symbol(0),
  list: Symbol(1),
});
