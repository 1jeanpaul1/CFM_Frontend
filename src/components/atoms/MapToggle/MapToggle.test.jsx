import MapToggle from './MapToggle';
import { screen, render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('MapToggle', () => {
  const defaultProps = {
    currentView: MapToggle.view.map,
    setView: jest.fn(),
  };

  const getRender = (props = defaultProps) => render(<MapToggle {...props} />);

  it('should render with correct text when currentView is map', () => {
    const { asFragment } = getRender();
    expect(screen.getByText('Map View')).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call setView function from props when button is clicked', () => {
    getRender();

    fireEvent.click(screen.getByText('Map View'));
    expect(defaultProps.setView).toHaveBeenCalledTimes(1);
  });
});
