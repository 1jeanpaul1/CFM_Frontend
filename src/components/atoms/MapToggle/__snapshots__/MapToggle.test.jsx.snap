// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapToggle should render with correct text when currentView is map 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1o4uu0l-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MapOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m20.5 3-.16.03L15 5.1 9 3 3.36 4.9c-.21.07-.36.25-.36.48V20.5c0 .28.22.5.5.5l.16-.03L9 18.9l6 2.1 5.64-1.9c.21-.07.36-.25.36-.48V3.5c0-.28-.22-.5-.5-.5zM10 5.47l4 1.4v11.66l-4-1.4V5.47zm-5 .99 3-1.01v11.7l-3 1.16V6.46zm14 11.08-3 1.01V6.86l3-1.16v11.84z"
        />
      </svg>
    </span>
    Map View
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
